### PR TITLE
docs: add Cloudflare Workers documentation

### PR DESCRIPTION
**Pull request by victoriabernard92**
closes https://github.com/gatsbyjs/gatsby/issues/21652

### CODE REVIEW

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+type = "webpack"
+account_id = "abcd..."
+workers_dev = true
+
+[site]
+bucket = "./public"
+entry-point = "workers-site"
+```
+
+This deploys to your workers.dev subdomain. For a custom domain see [Quick Start]().
```
Comment (signalnerve) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
This deploys to your workers.dev subdomain. For a custom domain see [Quick Start](https://developers.cloudflare.com/workers/quickstart/#publish-to-your-domain).
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      run: cd $GITHUB_WORKSPACE
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
+    - name: Install deps
+      run: npm install
+    - name: Build docs
+      run: npm run build
+    - name: Publish
+      uses: signalnerve/wrangler-action@1.0.0
```
Comment (signalnerve) [inline docs/docs/deploying-to-cloudflare-workers.md]: Maybe this is out-of-date in our docs, but the repo has moved, so we can do...
```suggestion
      uses: cloudflare/wrangler-action@1.1.0
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+wrangler publish
+```
+
+Now your site is available at gatsby-project.subdomain.workers.dev!
+
+## Step 4: CI with Github Actions
+
+Use wrangler's Github action [plugin](https://github.com/cloudflare/wrangler-action) to automatically deploy to Workers every time you push to master.
+
+Once Github actions is enabled on your repo add a file to projects root called `.github/workflows/main.yml` with the contents:
```
Comment (signalnerve) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
Once Github Actions is enabled on your repo, add a file to your project's root called `.github/workflows/main.yml` with the contents:
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      with:
+        node-version: '10.x'
+    - name: Install deps
+      run: npm install
+    - name: Build docs
+      run: npm run build
+    - name: Publish
+      uses: signalnerve/wrangler-action@1.0.0
+      with:
+        apiKey: ${{ secrets.CF_API_KEY }}
```
Comment (signalnerve) [inline docs/docs/deploying-to-cloudflare-workers.md]: We should swap this usage out with API tokens -

```
with:
  apiToken: ${{ secrets.CF_API_TOKEN }}
```

and update the below section on where to find it accordingly

File: docs/blog/100days/free-hosting/index.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ Bonus: Try deploying your site to the free tier of multiple services to see firs
 
 For your git repo, you can use GitHub, Gitlab, Bitbucket, or many similar services.
 
-And for your build tool and CDN we see many people have success with [AWS Amplify](/docs/deploying-to-aws-amplify/), [Netlify](/docs/deploying-to-netlify/), and [ZEIT Now](/docs/deploying-to-zeit-now/). Our documentation has step-by-step tutorials for those services, [and others](/docs/deploying-and-hosting/).
+And for your build tool and CDN we see many people have success with [AWS Amplify](/docs/deploying-to-aws-amplify/), [Netlify](/docs/deploying-to-netlify/), [ZEIT Now](/docs/deploying-to-zeit-now/), and [Cloudflare Workers](/docs/deploying-to-cloudflare-workers/),. Our documentation has step-by-step tutorials for those services, [and others](/docs/deploying-and-hosting/).
```
Comment (laurieontech) [inline docs/blog/100days/free-hosting/index.md]:
```suggestion
And for your build tool and CDN we see many people have success with [AWS Amplify](/docs/deploying-to-aws-amplify/), [Netlify](/docs/deploying-to-netlify/), [ZEIT Now](/docs/deploying-to-zeit-now/). Our documentation has step-by-step tutorials for those services, [and others](/docs/deploying-and-hosting/).
```

This blog should not be changed as a part of this PR.

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+## Step 1: Installing wrangler CLI
+
+Workers Sites requires [wrangler](https://developers.cloudflare.com/workers/tooling/wrangler/). The more straight forward way to install wrangler is with [npm](https://www.npmjs.com/), run the following command:
+
+```shell
+npm i -g @cloudflare/wrangler
+```
+
+## Step 2: Init Project
```
Comment (laurieontech) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
2. Initialize the Project
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
+---
+title: Deploying to Cloudflare Workers
+---
+
+[Cloudflare Workers](https://workers.cloudflare.com/) is a serverless platform for creating entirely new applications or augmenting existing ones without configuring or maintaining infrastructure. With [Workers Sites](https://developers.cloudflare.com/workers/sites/start-from-existing/) you can deploy any static site including your Gatsby projects to a domain on Cloudflare or for free with on your [workers.dev](workers.dev) subdomain.
+
+This guide will get you started in a few steps:
+
+## Step 1: Installing wrangler CLI
```
Comment (laurieontech) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
1. Installing wrangler CLI
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    ..
+├── src
+│   ├── components
+│   ├── images
+│   └── pages
+├── gatsby-config.js
+├── package.json
+```
+
+## Configure
```
Comment (laurieontech) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
3. Configure
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+workers_dev = true
+
+[site]
+bucket = "./public"
+entry-point = "workers-site"
+```
+
+This deploys to your workers.dev subdomain. For a custom domain see [Quick Start](https://developers.cloudflare.com/workers/quickstart/#publish-to-your-domain).
+
+## Step 3: Deploy
```
Comment (laurieontech) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
4. Deploy
```

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+You can deploy your application by running the following command in the root of the project directory:
+
+```shell
+wrangler publish
+```
+
+Now your site is available at gatsby-project.subdomain.workers.dev!
+
+## Step 4: CI with Github Actions
```
Comment (laurieontech) [inline docs/docs/deploying-to-cloudflare-workers.md]:
```suggestion
5. CI with Github Actions
```

File: www/src/data/tags-docs.js [hunk 11]
```diff
@@ -20,6 +20,7 @@ export const TAGS_AND_DOCS = new Map([
   [`cdn`, ``],
   [`cli`, ``],
   [`client-side-routing`, `/docs/client-only-routes-and-user-authentication`],
+  [`cloudflare`, "/docs/deploying-to-cloudflare-workers"],
```
Comment (gillkyle) [inline www/src/data/tags-docs.js]:
```suggestion
```

Since there aren't any blog posts tagged Cloudflare we don't need to add this here 🙂

**Comment by bball07** [comment]
Hello @victoriabernard92,

Great work on this! I'm working through the steps and running into some trouble with this error:

Error: Your user configuration is invalid, please run wrangler config and enter a new set of credentials.

I think I followed the steps as they were outlined here, is there anything I may be missing?

**Comment by victoriabernard92** [comment]
> Error: Your user configuration is invalid, please run wrangler config and enter a new set of credentials.

Sorry for your trouble. You will need to run `wrangler config` and then will be prompted for your Cloudflare API Token. Steps on how to generate your token are in [this article](https://support.cloudflare.com/hc/en-us/articles/200167836). Were you able to generate your token?

**Comment by victoriabernard92** [comment]
@LekoArts anything I can do to help on this?

File: docs/docs/deploying-to-cloudflare-workers.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+This deploys to your workers.dev subdomain. For a custom domain see [Quick Start](https://developers.cloudflare.com/workers/quickstart/#publish-to-your-domain).
+
+4. Deploy
+
+You can deploy your application by running the following command in the root of the project directory:
+
+```shell
+wrangler publish
```
Comment (gillkyle) [inline docs/docs/deploying-to-cloudflare-workers.md]: I got to this point okay and get the following error:

```
Error: You will need to enable Workers Unlimited for your account before you can use this feature.
```

Looking into this it seems like the `public` folder gets saved to Cloudflare's KV storage which requires a premium plan, is that correct? I didn't have one, maybe we can make it clear at the top of the article that you need a premium account? Perhaps I'm mistaken and there is a way to do this on a free tier but I haven't been able to find it.

File: www/src/data/sidebars/doc-links.yaml [hunk 13]
```diff
@@ -61,6 +61,9 @@
             - title: Deploying to Netlify
               link: /docs/deploying-to-netlify/
               breadcrumbTitle: Netlify
+            - title: Deploying to Cloudflare Workers
```
Comment (gillkyle) [inline www/src/data/sidebars/doc-links.yaml]: Could this be moved a bit further down? Other deployment solutions tend to be requested more common and the ordering of these docs tends to be with the most relevant first. Perhaps below Zeit Now, like the recipe you wrote is on the recipes page?

Comment (victoriabernard92) [reply www/src/data/sidebars/doc-links.yaml]: Done

File: docs/docs/deploying-to-cloudflare-workers.md (hunk 12 above)
Comment (victoriabernard92) [reply docs/docs/deploying-to-cloudflare-workers.md]: Done

**Pull request merged**